--- ./ecartis.cfg.dist.org	Tue Mar 26 08:53:01 2002
+++ ./ecartis.cfg.dist	Mon Apr  8 12:37:04 2002
@@ -32,10 +32,10 @@
 #    This is where to load the Ecartis modules from.   Defaults to
 #    listserver-root/modules
 #
-# listserver-root = /home/ecartis
-# listserver-conf = /home/ecartis
-# listserver-data = /home/ecartis
-# listserver-modules = /home/ecartis/modules
+listserver-root = /usr/lib/ecartis
+listserver-conf = /etc/ecartis
+listserver-data = /usr/lib/ecartis
+listserver-modules = /usr/lib/ecartis/modules
 
 # List directory
 # In the situation where you are virtual hosting, you could in fact
@@ -46,7 +46,7 @@
 lists-root = lists
 
 # Ecartis logfile location
-logfile = ./ecartis.log
+logfile = /var/log/ecartis.log
 
 # Paranoia (boolean)
 # Are we a paranoid installation?  Paranoid installations only allow
