--- ./src/inc/config.h.org	Tue Mar  4 08:45:05 2003
+++ ./src/inc/config.h	Tue Mar  4 22:01:50 2003
@@ -20,7 +20,7 @@
 #define SERVICE_NAME_UC "ECARTIS"
 #define SERVICE_NAME_MC "Ecartis"
 #define SERVICE_NAME_LC "ecartis"
-#define GLOBAL_CFG_FILE "ecartis.cfg"
+#define GLOBAL_CFG_FILE "/etc/ecartis/ecartis.cfg"
 #define SERVICE_ADDRESS "ecartis@localhost"
 
 #endif /* _CONFIG_H */
--- ./src/variables.c.org	Tue Mar  4 08:45:05 2003
+++ ./src/variables.c	Tue Mar  4 22:03:50 2003
@@ -1295,12 +1295,12 @@
     register_var("default-flags", "|ECHOPOST|", "Basic Configuration",
                  "Default flags given to a user when they are subscribed.",
                  "default-flags = |NOPOST|DIGEST|", VAR_STRING, VAR_ALL);
-    register_var("global-blacklist", "banned", "Files",
+    register_var("global-blacklist", "/etc/ecartis/banned", "Files",
                  "Global file containing regular expressions for users who are not allowed to subscribe to lists hosted on this server.",
-                 "global-blacklist = banned", VAR_STRING, VAR_GLOBAL|VAR_SITE);
+                 "global-blacklist = /etc/ecartis/banned", VAR_STRING, VAR_GLOBAL|VAR_SITE);
     register_var("logfile", NULL, "Debugging",
                  "Filename where debugging log information will be stored.",
-                 "logfile = ./server.log", VAR_STRING, VAR_GLOBAL|VAR_SITE);
+                 "logfile = /var/log/ecartis.log", VAR_STRING, VAR_GLOBAL|VAR_SITE);
     register_var("full-bounce", "no", "SMTP",
                  "Should bounces contain the full message or only the headers.",
                  "full-bounce = false", VAR_BOOL, VAR_ALL);
@@ -1317,7 +1317,7 @@
     register_var("moderate-verbose-subject", "yes", "Moderation",
                  "Should moderated messages have a more informative subject?",
                  "moderate-verbose-subject = yes", VAR_BOOL, VAR_ALL);
-    buffer_printf(buf, sizeof(buf) - 1, "%s.hlp", SERVICE_NAME_LC);
+    buffer_printf(buf, sizeof(buf) - 1, "/etc/ecartis/%s.hlp", SERVICE_NAME_LC);
     register_var("no-command-file", buf, "Files",
                  "This is a global file to send if a message to the main listserver or request address has no commands.",
                  "no-command-file = helpfile",VAR_STRING, VAR_GLOBAL|VAR_SITE);
